app/controllers/repositories_controller.rb

```diff
@@ -100,14 +103,21 @@ def branches
   end
 
   def notify_push
```

Comment:
I thought about that, but having a single action makes it easier for users and to migrate repositories between services.